OnPageCode.js

```diff
@@ -1,219 +1,411 @@
+
+      currentBidCall.order = sspConfig.bids[processedBidsCount - 1].order;
+      currentBidCall.reqMethod = sspConfig.bids[processedBidsCount - 1].reqMethod;
+      currentBidCall.timestamp = Date.now();
+      currentBidCall.nw = {
+        evaluated: {
+          // percieved latency
+          per_lt: Date.now() - PM_Network_POC.cache.auction.timestamp
+        },
         raw: {}
-      },
-      serverTimelines: timelines
-    };
+      };
+      currentBidCall.requestUrlPayloadLength = bidderRequest?.nwMonitor?.requestUrlPayloadLength || sspConfig.bids[processedBidsCount - 1].reqPayloadCharCount;
 
-    output.bidder = {
-      name: sspConfig.name,
-      key: sspConfig.key
-    };
+      PM_Network_POC.statHatParameters.forEach(parameter => {
+        const value = PM_Network_POC.getTimeValue(
+          perfResource[parameter.timeEndKey],
+          perfResource[parameter.timeStartKey]
+        );
+        if (value > 0) {
+          currentBidCall.nw.evaluated['nw_' + parameter.key] = value;
+        }
+      });
+
+      const jsonPerfResource = JSON.stringify(perfResource);
+      const raw = JSON.parse(jsonPerfResource);
+      raw.name = PM_Network_POC.removeQueryParams(perfResource.name);
+      currentBidCall.nw.raw = raw;
+      PM_Network_POC.bidCallTracker.push(currentBidCall);
 
-    PM_Network_POC.statHatParameters.forEach(parameter => {
-      const value = PM_Network_POC.getTimeValue(
-        perfResource[parameter.timeEndKey],
-        perfResource[parameter.timeStartKey]
-      );
-      if (value > 0) {
-        output.nw.evaluated['nw_' + parameter.key] = value;
+      if (processedBidsCount === bidderRequest.bids.length) {
+        PM_Network_POC.uploadTheNetworkLatencyData(output);
+        PM_Network_POC.bidCallTracker = [];
       }
-    });
-
-    output.nw.raw = perfResource;
-    PM_Network_POC.uploadTheNetworkLatencyData(output);
-  },
-
-  performNetworkAnalysis: function (timelines, uniqueReqId) {
-    let performanceResources = window?.performance?.getEntriesByType("resource");
-    let i = PM_Network_POC.lastExecutionMaxIndex;
-    for (; i < performanceResources.length; i++) {
-      let perfResource = performanceResources[i];
-      PM_Network_POC.lastExecutionMaxIndex++;
-
-      let sspConfig = PM_Network_POC.bidders.find(
-        bidder => perfResource.name.includes(bidder.searchName)
-      ) || null;
-
-      if (sspConfig) {
-        if(sspConfig.key === 'pm') {
-          const value = PM_Network_POC.getParameterByName("uniqueReqId", perfResource.name);
-          if(value == uniqueReqId) {
-            PM_Network_POC.prepareNetworkLatencyData(perfResource, sspConfig, timelines);
+    },
+
+    removeQueryParams: url => url.split('?')[0],
+
+    isPubMaticBidder: function (bidderCode) {
+      return bidderCode === 'pubmatic';
+    },
+
+    getBidderByBidRequest: function (bidderRequest) {
+      return PM_Network_POC.bidders.find(bidder => bidder.bidderCode === bidderRequest.bidderCode);
```

Comment:
I removed the `bidderRequest?.nwMonitor?.reqMethod` check from following if statement since we can receive a bidder's bid request method dynamically from pbjs
```
if (PM_Network_POC.isPubMaticBidder(bidder.bidderCode)) {
  return bidder.bidderCode === bidderRequest.bidderCode && bidder.reqMethod === bidderRequest?.nwMonitor?.reqMethod;
}
```

is this ok?

Reply:
Yes, Please check the latest code from `OnPageCode_timeout.js ` from the branch `ns/exp/uoe-8525` as I have updated this function along with many other changes. 
I think you should go through it and if needed we can discuss it, to make it better.

Reply:
yes, this pr i created was on top of your latest changes in `OnPageCode_timeout.js` from the `s/exp/uoe-8525` branch.  i think it would be safe to rely on the req method that pbjs returns since all bid request methods are captured

Reply:
ah, just seeing now that as i'm reading through this that more commits to your branch were recently made.  i will check those out to get up to speed

Reply:
Yes, please!